src/Microsoft.SqlTools.ServiceLayer/LanguageServices/LanguageService.cs

```diff
@@ -452,11 +453,15 @@ internal async Task HandleSyntaxParseRequest(SyntaxParseParams param, RequestCon
                 }
                 else
                 {
-                    // get the current list of completion items and return to client
-                    ConnectionServiceInstance.TryFindConnection(
+                    ConnectionInfo connInfo = null;
+                    if (!await connectionService.TryRefreshAuthToken(scriptFile.ClientUri))
+                    {
+                        // if the auth token needs to be refreshed, we will disable intellisense until it has been refreshed.
```

Comment:
The comment being in here is confusing - unless I'm misunderstanding this being inside the if block here means that we didn't need to refresh the auth token.

I'd instead suggest removing this and just put a comment above the if 

`Check if we need to refresh the auth token, and if we do then don't pass in the connection so that we only show the default options until the refreshed token is returned`